app/models/king.rb

```diff
@@ -1,8 +1,12 @@
 class King < Piece
-  # ensures king piece can only move 1 square in any direction
+
   def valid_move?(x,y)
-    # checks critera for valid_move? that apply to all pieces and returns false if fails
-    return false unless super(x,y)
+    # checks critera for valid_move? that apply to all pieces and returns false if fails unless can_castle
+    (normal_valid_move?(x,y) && super(x,y)) || can_castle?(x,y)
+  end
+
+  # ensures king piece can only move 1 square in any direction
```

Comment:
this is what was there before my changes:
class King < Piece
  # ensures king piece can only move 1 square in any direction
  def valid_move?(x,y)
    # checks critera for valid_move? that apply to all pieces and returns false if fails
    return false unless super(x,y)
    x_diff = x_diff(x)
    y_diff = y_diff(y)
    # returns true if moving only 1 square any direction
    if (x_diff <= 1) && (y_diff <= 1)
      return true
    end
    # returns false if not moving only 1 square any direction
    return false
  end

I moved the logic you mentioned to "normal valid move" and adjusted the tests for that as well.  It's still there, just rearranged to allow for castling.